src/page-modules/assistant/trip/trip-pattern/index.tsx

```diff
@@ -71,6 +71,18 @@ export default function TripPattern({
     [style['tripPattern--old']]: tripIsInPast,
   });
 
+  const staySeated = (idx: number) =>
+    expandedLegs[idx].interchangeTo?.staySeated === true;
+
+  const shouldHaveRemainedSeatedInPreviousLeg = (idx: number) => {
+    if (idx > 0) return staySeated(idx - 1);
+    return false;
+  };
```

Comment:
With this usage (further down) you won't have cases where expandedLegs[idx] is undefined, due to checking for it a different place (in shouldHaveRemainedSeatedInPreviousLeg). But the check in shouldHaveRemainedSeatedInPreviousLeg is dependent on internal knowledge of staySeated. So you are kind of mixing knowledge of implementations and spreading it across different functions. No real practical consequences in this case, but it can be seen as a little "un clean" in a sense. And over time these functions could be moved, changed, etc, which introduces this as a problem. It can be better to be somewhat more paranoid when implementing functions like these. As to assume that no consumer has done checks/guards for you. So you would have to do null checks in staySeated also

Reply:
Makes sense! I agree that splitting the logic between several functions can make the logic a bit unclear. I also now see that by adding  a null checker to `staySeated`, `shouldHaveRemainedSeatedInPreviousLeg` becomes redundant as `staySeated` will return false as a result of the null checker, in case the leg is undefined.